userspace/libsinsp/runc.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright (C) 2013-2019 Draios Inc dba Sysdig.
+
+This file is part of sysdig.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "runc.h"
+
+#include <cstring>
+
+#include "sinsp.h"
+#include "sinsp_int.h"
+
+namespace {
+
+const size_t CONTAINER_ID_LENGTH = 64;
+const size_t REPORTED_CONTAINER_ID_LENGTH = 12;
+const char* CONTAINER_ID_VALID_CHARACTERS = "0123456789abcdefABCDEF";
+
+static_assert(REPORTED_CONTAINER_ID_LENGTH <= CONTAINER_ID_LENGTH, "Reported container ID length cannot be longer than actual length");
+
+bool detect_one_container_id(const std::string& cgroup, const std::string& prefix, const std::string& suffix, std::string& container_id)
+{
+	size_t pos = cgroup.rfind(prefix);
+	if (pos == std::string::npos)
+	{
+		return false;
+	}
+	pos += prefix.size();
+
+	size_t pos2 = cgroup.rfind(suffix);
```

Comment:
this check is iffy too....what if the scope appears somewhere earlier in the string and we're not expecting it?

why not just string compare the last strlen(suffix) chars of the string with suffix?

Reply:
plus rfind is expensive...as we have to search the whole string if it doesn't exist

Reply:
is there any reason we're not just using regex for this stuff?

Reply:
(see above^2 comment). We can't use regexes as we don't have a suitable library handy (stdlib regexes are broken in gcc 4.8).

We probably pull in pcre already in one of the dependencies but I don't think we use it directly.

Reply:
is there any aversion to poco regex? Is it not used in sysdig at all?

either way, any reason not to just directly check for the suffix rather than using rfind?